hack/benchmark/core/benchmark/strategy/bulk_insert.go

```diff
@@ -26,18 +26,62 @@ import (
 	"github.com/vdaas/vald/hack/benchmark/internal/core"
 )
 
+const (
+	maxBulkSize = 100000
+)
+
 func NewBulkInsert(opts ...StrategyOption) benchmark.Strategy {
 	return newStrategy(append([]StrategyOption{
 		WithPropName("BulkInsert"),
 		WithProp32(
 			func(ctx context.Context, b *testing.B, c core.Core32, dataset assets.Dataset, ids []uint, cnt *uint64) (interface{}, error) {
-				ids, errs := c.BulkInsert(dataset.Train())
+				size := func() int {
+					if maxBulkSize < dataset.TrainSize() {
+						return maxBulkSize
+					} else {
+						return dataset.TrainSize()
+					}
+				}()
+				v := make([][]float32, 0, size)
+				for i := 0; i < size; i++ {
+					arr, err := dataset.Train(i)
+					if err != nil {
+						break
```

Comment:
will we miss some data if we break the loop here?

Reply:
In almost cases err is nil because Train function returns only ErrOutOfBounds as error and size is less or equal train dataset size.
I think this sentence is redundant now, but Train may changes returning other error.
So that, it might be better dividing the process that err is ErrOutOfBounds and the others.

Reply:
I mean the data may not be loaded completely whenever the error is occurred, so I think there is a case that the error occurred and the `arr` is not append to `v` in [line 51](https://github.com/vdaas/vald/pull/612/files#diff-408799d8c77478a76c2fa1f9d331d158R51).

What do you think?

Reply:
Sorry, I didn't think seriously about it. You are right.
The best way I think is changing WithProp32 interface like this and WithProp64 in the same way.
`func(ctx context.Context, b *testing.B, c core.Core32, [][]float32, ids []uint, cnt *uint64) (interface{}, error)`
The good point is what enables to use same [][]float32 data in all benchmarks. We can get fair benchmark in the same benchmark using the same data, if error is occurred. (However, We may get the unfair result in the different benchmark.)
The bad point is what changes interfaces and needs to implement data generation. It could be a complex implementation.

Reply:
I am not sure if we can do this? to train the dataset from index 0 again whenever error occurred 
```go
				for i := 0; i < size;  {
					arr, err := dataset.Train(i % size)
					if err == nil {
					         v = append(v, arr.([]float32))
                                                  i++
					}
				}
```

Reply:
what I concerned was that the testing result may inaccurate when error is occurred

Reply:
Sorry, I sent it halfway through.
If my proposal is chosen, I think that it is good to change it in the new pull request.

Reply:
Sorry, I have reconsidered.
Benchmark is important to use same data for comparing result.
So, my implementation is bad, and it might be good to stop the benchmark if error is occurred.
How about it?

Reply:
`it might be good to stop the benchmark if error is occurred`
I think it is ok :) 👍

Reply:
Thank you for your comment! I will fix to stop the benchmark.